<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find exploration)/cfg/custom_rosconsole.config"/>
    <!-- sensor-based -->
    <!-- <arg name="branch"/> sub topic -->
    <!-- <arg name="pose"/> sub topic -->
    <arg name="pose_log"/> <!-- sub topic -->

    <!-- seamless hybrid -->
    <!-- <arg name="robot_array"/> sub topic (header) -->
    <arg name="frontier" default="frontier"/> <!-- sub topic (header) -->
    <arg name="end"/> <!-- sub topic (main) -->
    <arg name="robot_name"/> <!-- param (header) -->
    <arg name="debug"/> <!-- param (main) -->
    <arg name="rotation"/> <!-- param (main) -->

    <!-- movement -->
    <!-- <arg name="scan"/> sub topic -->
    <!-- <arg name="bumper"/> sub topic -->
    <!-- <arg name="velocity"/> pub topic -->
    <!-- <arg name="pose"/> sub topic -->

    <!-- path planner (seamless hybrid) -->
    <arg name="map"/> <!-- sub topic -->
    <arg name="map_frame_id"/> <!-- param -->

    <!-- launch -->
    <!-- <arg name="debug" default="false"/> -->
    <!-- <arg name="robot_name"/> -->

    <arg name="area_diff"/> <!-- sub topic -->

    <node pkg="exploration" type="seamless_hybrid_exploration_actively_frontier" name="seamless_hybrid_exploration_actively_frontier" output="screen">
        <remap from="area_diff" to="$(arg area_diff)"/>
        <remap from="pose_log" to="$(arg pose_log)"/>
        <remap from="frontier" to="$(arg frontier)"/>
        <remap from="end" to="$(arg end)"/>
        <remap unless="$(arg debug)" from="velocity" to="/$(arg robot_name)/mobile_base/commands/velocity"/>
        <remap from="map" to="$(arg map)"/>
        <remap from="mobile_base/sensors/bumper_pointcloud" to="/$(arg robot_name)/mobile_base/sensors/bumper_pointcloud"/>
        <remap from="global_costmap" to="/$(arg robot_name)/move_base/global_costmap/costmap"/>

        <rosparam file="$(find exploration)/param/seamless_planner_params.yaml" command="load"/>
        <param name="seamless_costmap/global_frame" value="$(arg map_frame_id)"/>
        <param name="seamless_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint"/>

        <rosparam file="$(find exploration)/param/movement_planner_params.yaml" command="load"/>
        <param name="movement_costmap/global_frame" value="$(arg map_frame_id)"/>
        <param name="movement_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint"/>
        
        <param name="rotation" value="$(arg rotation)"/>
        <param name="debug" value="$(arg debug)"/>

        <rosparam file="$(find exploration)/param/movement_she_last_parameters.yaml" command="load" ns="movement"/>
        <param name="movement/movement_parameter_file_path" value="$(find exploration)/param/movement_she_last_parameters.yaml"/>
        <param name="movement/movebase_name" value="/$(arg robot_name)/move_base"/>
        
        <rosparam file="$(find exploration)/param/she_last_parameters.yaml" command="load" ns="seamless_hybrid_exploration"/>
        <param name="seamless_hybrid_exploration/she_parameter_file_path" value="$(find exploration)/param/she_last_parameters.yaml"/>
        <param name="seamless_hybrid_exploration/robot_name" value="$(arg robot_name)"/>
        
        <param name="sensor_based_exploration/output_sbe_parameters" value="false"/>
    </node>
</launch>
